lib/providers/map_data.dart

```diff
@@ -1,113 +1,110 @@
 import 'dart:async';
-import 'package:concordia_navigation/services/itinerary.dart';
+import 'package:concordia_navigation/models/reachable.dart';
+import 'package:concordia_navigation/models/uni_location.dart';
+import 'package:concordia_navigation/services/outdoor/location_service.dart';
+import 'package:concordia_navigation/services/outdoor/outdoor_itinerary.dart';
 import 'package:flutter/foundation.dart';
 import 'package:google_maps_flutter/google_maps_flutter.dart';
 import 'package:flutter/material.dart';
 import 'package:sliding_up_panel/sliding_up_panel.dart';
+import 'package:concordia_navigation/storage/app_constants.dart' as constants;
 
 ///Observer Pattern
 ///Handles all the data related to the map, listens to changes and notifies listeners.
 class MapData extends ChangeNotifier {
   Completer<GoogleMapController> _completer = Completer();
   PanelController panelController = new PanelController();
-  String controllerStarting;
-  String controllerDestination;
-  double swapButtonTop;
-  double locationButtonTop;
+  LocationService locationService = LocationService.getInstance();
+  Reachable _start, _end;
+  bool panelVisible = false;
 
-  Itinerary itinerary;
+  String controllerStarting, controllerEnding;
+
+  OutdoorItinerary itinerary;
 
   Completer<GoogleMapController> get getCompleter {
     return _completer;
   }
 
-  LatLng _currentLocation;
-  String _campus;
-  LatLng _start;
-  LatLng _end;
-  String _mode;
+  String mode;
 
   MapData() {
-    _mode = "driving";
+    mode = "driving";
   }
 
-  void changeCampus(campus) {
-    _campus = campus;
-    notifyListeners();
-  }
+  Reachable get start => _start;
+  Reachable get end => _end;
 
-  void changeSwapTop(double top) {
-    swapButtonTop = top;
+  // set the end Reachable object and use its name
+  set end(Reachable obj) {
+    _end = obj;
+    if (obj != null) controllerEnding = (obj as UniLocation).name;
+    else controllerEnding = null;
     notifyListeners();
   }
 
-  void changeLocationTop(double top) {
-    locationButtonTop = top;
+  // set the start Reachable object and use its name
+  set start(Reachable obj) {
+    _start = obj;
+    if (obj != null) controllerStarting = (obj as UniLocation).name;
+    else controllerStarting = null;
     notifyListeners();
```

Comment:
Following general coding conventions you might want to add curly brackets to your if statements for clarity.

Reply:
@WindOfRussia  Dart style guide says they're optional
https://dart.dev/guides/language/effective-dart/style#do-use-curly-braces-for-all-flow-control-structures
The curly braces can be omitted if the statement is one line.

Our own style guide on the wiki doesn't mention it. Should we make curly braces required or leave them up to preference still?

Reply:
We've been following this for a while now, one line if statements i mean. I'm closing this.